assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/blip/mlflow_wrapper.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""MLflow PythonModel wrapper class that loads the MLflow model, preprocess inputs and performs inference."""
+
+import mlflow
+from PIL import Image
+import pandas as pd
+import torch
+import tempfile
+
+from transformers import AutoProcessor, Blip2ForConditionalGeneration
+from config import MLflowSchemaLiterals, MLflowLiterals, Tasks
+from typing import List, Tuple
+
+try:
+    # Use try/except since vision_utils is added as part of model export and not available when initializing
+    # model wrapper for save_model().
+    from vision_utils import create_temp_file, process_image, get_current_device
+except ImportError:
+    pass
+
+
+class BLIPMLFlowModelWrapper(mlflow.pyfunc.PythonModel):
+    """MLflow model wrapper for BLIP model."""
+
+    def __init__(
+        self,
+        task_type: str,
+    ) -> None:
+        """Initialize MLflow wrapper class.
+
+        :param task_type: Task type used in training.
+        :type task_type: str
+        """
+        super().__init__()
+        self._processor = None
+        self._model = None
+        self._task_type = task_type
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """
+        Load a MLflow model with pyfunc.load_model().
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        if self._task_type == (Tasks.IMAGE_CAPTIONING.value):
+            try:
+                model_dir = context.artifacts[MLflowLiterals.MODEL_DIR]
+                self._processor = AutoProcessor.from_pretrained(model_dir)
+                self._model = Blip2ForConditionalGeneration.from_pretrained(
+                    model_dir)
+                self._device = get_current_device()
+                self._model.to(self._device)
+
+                print("Model loaded successfully")
+            except Exception as e:
+                print("Failed to load the the model.")
+                print(e)
+                raise
+        else:
+            raise ValueError(f"invalid task type {self._task_type}")
+
+    def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.DataFrame) -> pd.DataFrame:
+        """Perform inference on the input data.
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        :param input_data: Input images for prediction and text for the question.
+        :type input_data: Pandas DataFrame with a first column name ["image"] of images where each
```

Comment:
Is this docstring accurate? From the config.py it seems we only have one input column?

Reply:
fixed